### PR TITLE
Add document generation test

### PR DESCRIPTION
**Pull request by kouml**
### Summary
We should try to build the docs in travis to catch document generation errors.

### Related Issues
https://github.com/keras-team/keras/pull/11880

### PR Overview
- Add UTF-8 and mkdocs in Dockerfile, travis.yml
- Test `python autogen.py && mkdocs build`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -54,10 +54,9 @@ install:
 
   # install TensorFlow (CPU version).
   - pip install tensorflow==1.9 --progress-bar off
-  
+
   # install cntk
   - if [[ "$KERAS_BACKEND" == "cntk" ]] || [[ "$TEST_MODE" == "PEP8_DOC" ]]; then
-      ./.travis/install_cntk.sh;
```
Comment (gabrieldemarmiesse) [inline .travis.yml]: What do you have against CNTK? :p

File: .travis.yml [hunk 2]
```diff
@@ -69,6 +69,12 @@ install:
   - export APP_CHANGED=False;
   - for entry in `git diff --name-only HEAD~1`; do if [[ "$entry" == "keras/applications/"* ]]; then export APP_CHANGED=True; fi; done
 
+  # set utf-8
+  - export LC_ALL=C.UTF-8;
+  - export LANG=C.UTF-8;
```
Comment (gabrieldemarmiesse) [inline .travis.yml]: It seems that travis already uses UTF8 by default. See https://docs.travis-ci.com/user/environment-variables/#default-environment-variables .

I don't think that those two lines are useful.

File: .travis.yml (hunk 1 above)
Comment (kouml) [reply .travis.yml]: That is my mistake... I'm sorry about that.

File: .travis.yml (hunk 2 above)
Comment (kouml) [reply .travis.yml]: Thanks for reviewing! I checked and removed it.

File: .travis.yml (hunk 1 above)
Comment (gabrieldemarmiesse) [reply .travis.yml]: No problem :)

File: .travis.yml (hunk 2 above)
Comment (gabrieldemarmiesse) [reply .travis.yml]: Thanks!

**Pull request merged**